v2/controllers/pod_watcher.go

```diff
@@ -55,6 +58,37 @@ func SetupPodWatcher(mgr ctrl.Manager, ns, name string, ft founat.FoUTunnel, enc
 		peers:          make(map[string]map[string]struct{}),
 	}
 
+	if cfg == nil {
+		cfg = config.GetConfigOrDie()
+	}
+
+	client, err := client.New(cfg, client.Options{})
+	if err != nil {
+		return err
+	}
+
+	var pods corev1.PodList
+	if err := client.List(context.Background(), &pods); err != nil {
```

Comment:
How about declaring `ctx`?
If we do this, we can use same `ctx` at Line 85.
```suggestion
        ctx := context.Background()
	if err := client.List(ctx, &pods); err != nil {
```

Reply:
I fixed it, thanks :)